### DISCUSSION TITLE
Duplicate Superset Dasshboard via Import/Export and ability to point to new DataSet

### DISCUSSION THREAD

**Discussion by aidatabases** [apache/superset#30288] [Ideas]
Possible Use Case:
ESG (Environmental, Social, and Governance) Initiative related Dashboards for different Organizations.

Once I create 1 Dashboard for **Organization X**, I would like to replicate the same (CSS Styling/Positioning, Static and MD Content, Charts and Tabs etc ) for **Organization Y, Z** and many more.

Today, I can clone the existing Dashboard and duplicate Charts inside, but I cannot point it to a new Dataset.
I may have to edit every Duplicate chart and change the Dataset manually.
Would love if through Import/Export just by changing the Dataset UUID this requirement can be achieved

**Comment by yousoph** [comment]
Hi @aidatabases ! Did you try changing the dataset UUID in the export file? That may work if you already have your second dataset created and know the UUID.

**Comment by HeathLee** [comment]
We need this too. Usually, we have dev and production environment and build chart or dashboard first, and then want to copy them to prod easily

**Comment by SharonCastel** [comment]
Hi, When I export+import dashboard from one environment to another I would like it to overwrite all but even though I asked to overwrite I'm getting an error on duplicated key uuid for some charts and for others not. is there a logic here? how can I have a clean dashboard import including datasets and charts?

**Comment by AzeemIqbal** [comment]
We would love this too, same use case, dashboards which need to be viewed by multiple organisations, either the ability to clone the dashboard and point to filtered datasets, or create users and lock a filter on a dataset/dashboard for them

**Comment by aparlapalli** [comment]
Would also like to bump this. Looked into the UUIDs and I'm not sure if we need to generate new UUIDs for all the charts in the charts folder. If we do, that would require additional scripting to generate new UUIDs and I'm not sure if that would need to match the way Superset generates UUIDs for the charts. If there is a way to call / get the UUIDs for each dataset in our superset instance and only have to update that, would be worth knowing.

**Comment by GuyAttia** [comment]
This issue is very important for us as well. We are having very hard time to maintain the same dashboards in 2 environments (for dev-prod mechanism)

**Comment by joannren** [comment]
Resonant with above comments! we also have multiple environments, internal dev/prod, external dev/prod to maintain the same dashboard. Some dashboard(s) tie to dataset(s) with many metrics, calculated columns and 100 charts or more.  Duplicate such dashboards in different environments is very hard.

**Comment by 3mmdrew** [comment]
Currently attempting to deploy prod/stage/dev supersets for internal company data. Some supported feature for env replication or at least exports with variable dataset connections based on env variables would be incredibly useful!
The overwriting of the database files in the export is feasible but feels like a clunky and risky way of deploying.

**Comment by kleky** [comment]
A +1 from us too. Seem like this would be a step toward making superset enterprise grade. Right now it doesn't support typical product test/releases processes.

**Comment by rusackas** [comment]
Has anyone here tried what @yousoph suggested? 

Moving this to Discussion since I think that's the right place to carry this forward, since it's not really a bug report.

**Reply by GuyAttia** [reply]
Yes, we tried this approach, it's quickly turn into something very complicated because there are references for Datasets and Charts and DBs based on the local env UUID that need to be replaced as well

**Comment by StephanPF** [comment]
Same here, we need to deploy the same dashboard for different organisations that have the same data structure. Everything remain the same, dashboard, charts etc.. we just need to swap the dataset. Would be very useful also for deploying a staging environment.
